internal/cmd/cli/cleanup/cleanup.go

```diff
@@ -0,0 +1,113 @@
+package cleanup
+
+import (
+	"context"
+	"time"
+
+	"github.com/jpillora/backoff"
+	"github.com/rancher/fleet/internal/client"
+	"github.com/sirupsen/logrus"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type Options struct {
+	Min    int
+	Max    int
+	Factor float64
+}
+
+func ClusterRegistrations(ctx context.Context, client *client.Getter, opts Options) error {
+	c, err := client.Get()
+	if err != nil {
+		return err
+	}
+
+	// get the clients
+	cluster := c.Fleet.Cluster()
+	clusterRegistration := c.Fleet.ClusterRegistration()
+	serviceAccount := c.Core.ServiceAccount()
+	role := c.RBAC.Role()
+	roleBinding := c.RBAC.RoleBinding()
+	clusterRoleBinding := c.RBAC.ClusterRoleBinding()
+
+	// lookup for all existing clusters
+	seen := map[types.NamespacedName]struct{}{}
+	clusterList, _ := cluster.List("", metav1.ListOptions{})
+	for _, c := range clusterList.Items {
+		clusterKey := types.NamespacedName{Namespace: c.Namespace, Name: c.Name}
+		seen[clusterKey] = struct{}{}
+	}
+
+	crList, _ := clusterRegistration.List("", metav1.ListOptions{})
+
+	logrus.Debugf("Found %d clusters and %d cluster registrations", len(clusterList.Items), len(crList.Items))
+
+	latestGranted := map[types.NamespacedName]metav1.Time{}
+	for _, cr := range crList.Items {
+		// delete cluster registrations that have no cluster
+		clusterKey := types.NamespacedName{Namespace: cr.Namespace, Name: cr.Status.ClusterName}
+		if _, found := seen[clusterKey]; !found {
+			if err := clusterRegistration.Delete(cr.Namespace, cr.Name, nil); err != nil && !apierrors.IsNotFound(err) {
+				logrus.Errorf("Failed to delete cluster registration %s/%s: %v", cr.Namespace, cr.Name, err)
+			}
+			continue
+		}
+
+		// figure out the latest granted registration request per cluster
+		ts := latestGranted[clusterKey]
+		if cr.Status.Granted && ts.Before(&cr.CreationTimestamp) {
+			latestGranted[clusterKey] = cr.CreationTimestamp
+		}
+	}
+
+	// sleep at max 10 seconds to not overload the API server
+	b := &backoff.Backoff{
+		Min:    time.Duration(opts.Min) * time.Millisecond,
+		Max:    time.Duration(opts.Max) * time.Second,
+		Factor: opts.Factor,
+		Jitter: true,
+	}
+
+	go func() {
+		// it should be safe to delete up to the latest granted registration
```

Comment:
This looks less ambiguous to me ;)
```suggestion
		// it should be safe to delete up to, but not including, the latest granted registration
```